agr_api/src/main/java/org/alliancegenome/api/rest/interfaces/DiseaseRESTInterface.java

```diff
@@ -230,12 +232,12 @@ Response getDiseaseAnnotationsDownloadFile(@Parameter(in = ParameterIn.PATH, nam
 	@Operation(description = "Retrieve all DiseaseAnnotation records for a given disease id disregarding sorting / filtering parameters", hidden = true)
 	String getDiseaseAnnotationsDownload(@PathParam("id") String id);
 
-	@GET
+	@POST
 	@Path("")
 	@Operation(summary = "Retrieve all disease annotations of a given set of genes")
 	@APIResponses(value = { @APIResponse(responseCode = "200", content = @Content(mediaType = "application/json", schema = @Schema(implementation = Null.class))) })
 	JsonResultResponse<GeneDiseaseAnnotationDocument> getDiseaseAnnotationsRibbonDetails(@Parameter(in = ParameterIn.QUERY, name = "focusTaxonId", description = "Focus Taxon ID", required = true) @QueryParam("focusTaxonId") String focusTaxonId,
-		@Parameter(in = ParameterIn.QUERY, name = "geneID", description = "Gene by ID", required = true) @QueryParam("geneID") List<String> geneIDs, @Parameter(in = ParameterIn.QUERY, name = "termID", description = "Term ID by which rollup should happen") @QueryParam("termID") String termID,
+		@Parameter(in = ParameterIn.QUERY, name = "geneID", description = "Gene by ID", required = true) @RequestBody List<String> geneIDs, @Parameter(in = ParameterIn.QUERY, name = "termID", description = "Term ID by which rollup should happen") @QueryParam("termID") String termID,
```

Comment:
Sorry I know this isn't your code but the @Parameter all should be on a line by themselves. And the @RequestBody needs to be the last parameter to the method.